### PR TITLE
Fix writing of updated copyused scopes

### PR DESCRIPTION
**Pull request by sbomer**
Fixes https://github.com/mono/linker/issues/2138

### CODE REVIEW

**Review by vitek-karas — APPROVED** [review]
Looks good to me - @mateoatr please take a look as well (since you actually know the copyused better than me)

File: src/linker/Linker.Steps/SweepStep.cs [hunk 1]
```diff
@@ -696,12 +711,17 @@ void UpdateTypeScope (Collection<ExportedType> forwarders)
 					if (td == null) {
 						// Forwarded type cannot be resolved but it was marked
 						// linker is running in --skip-unresolved true mode
+						var anr = (AssemblyNameReference) f.Scope;
+						f.Scope = importer.ImportReference (anr);
```
Comment (mateoatr) [inline src/linker/Linker.Steps/SweepStep.cs:715]: Why is this needed?

Comment (sbomer) [reply src/linker/Linker.Steps/SweepStep.cs:715]: This was for consistency with how we treat unresolved typereferences here:
https://github.com/mono/linker/blob/73fbf96cb452f291cceba684034b8548d54d82eb/src/linker/Linker.Steps/SweepStep.cs#L908-L918

I think continue is right since we may be able to resolve other forwarders if not this one.

**Review by mateoatr — APPROVED** [review]

**Comment by odhanson** [comment]
@vitek-karas @sbomer are we planning to merge this into preview 7 ?

**Comment by vitek-karas** [comment]
As far as I can tell this is in Preview 7.
